### PR TITLE
Changes how reagents are added to victim of venomous component

### LINKED ISSUE #85386: Bees with advanced mutation toxin don't slimeify
## Issue Summary
image

## Reproduction:

Put advanced slime toxin in hive head

have it inject people

IT DOESNT WORK????????????????

on screenshot above everyone in shuttle was hit and tehy HAD OVER A HUNDRED UNITS and NONE GOT SLIMED?
image

This needs anything but touch. I don't think bees with the venomous element are meant to NOT INJECT.

### PR DESCRIPTION
**Pull request by Jewelry-x**
## About The Pull Request

This PR changes how reagents are added from the venomous component. Currently the component uses `add_reagents` which adds the reagents instantly to the target. 

While this may not seem like a problem, quite a lot of reagents require the `COMSIG_REAGENT_EXPOSE_MOB` for them to function, so a lot of reagents do not act as expected, such as in the issue #85386.

* Fixes #85386.
## Why It's Good For The Game

Makes reagents work as intended using the venomous component.
## Changelog
:cl:
fix: fixed reagents not being applied correctly by venomous mobs
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents.dm [hunk 1]
```diff
@@ -121,7 +121,7 @@
 	SHOULD_CALL_PARENT(TRUE)
 
 	. = SEND_SIGNAL(src, COMSIG_REAGENT_EXPOSE_MOB, exposed_mob, methods, reac_volume, show_message, touch_protection)
-	if((methods & penetrates_skin) && exposed_mob.reagents) //smoke, foam, spray
+	if(((penetrates_skin & methods) || (methods & INJECT)) && exposed_mob.reagents) //smoke, foam, spray
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents.dm:124]:
```suggestion
	if(penetrates_skin & (methods|INJECT)) //methods being 
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Jewelry-x** [comment]
After looking at the suggested change (which I blindly commited), correct me if I am wrong, but I think it is wrong, because I want to make sure the code runs if in methods exists INJECT, because penetrates_skin is solely for vapor foam and smoke. I have the impression the suggested change does not do this.

**Comment by Ghommie** [comment]
It isn't wrong, ((x|y) & z) is the same as (x & (y|z)) because of the rules of replacement that apply to boolean algebra.

**Comment by Jewelry-x** [comment]
> It isn't wrong, ((x|y) & z) is the same as (x & (y|z)) because of the rules of replacement that apply to boolean algebra.

When I tried in my local instance it did not work as intended though, so I am not sure.

Reagents were not added with the way you mentioned.

**Comment by Ghommie** [comment]
Actually nevermind. I'm a bit woozy and you are right.

**Comment by Jewelry-x** [comment]
> It should be (methods & (penetrates_skin|INJECT)).

The PR should be good then.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**